src/validator/IndexScanValidator.cpp

```diff
@@ -83,15 +97,148 @@ Status IndexScanValidator::prepareYield() {
+        return relExpr;
+    } else {
+        return genOrFilterFromList(tsExpr, vRet.value());
+    }
+    return nullptr;
+}
+
+std::unique_ptr<Expression> IndexScanValidator::genOrFilterFromList(
+    TextSearchExpression* expr, const std::vector<std::string>& values) {
+    auto filter = std::make_unique<LogicalExpression>(
+        Expression::Kind::kLogicalOr,
+        new RelationalExpression(
+            Expression::Kind::kRelEQ,
+            new LabelAttributeExpression(new LabelExpression(*expr->arg()->from()),
+                                         new LabelExpression(*expr->arg()->prop())),
+            new ConstantExpression(Value(values[0]))),
+        new RelationalExpression(
+            Expression::Kind::kRelEQ,
+            new LabelAttributeExpression(new LabelExpression(*expr->arg()->from()),
+                                         new LabelExpression(*expr->arg()->prop())),
+            new ConstantExpression(Value(values[1]))));
+    for (size_t i = 2; values.size() > 2 && i < values.size(); i++) {
+        auto left = new LogicalExpression(Expression::Kind::kLogicalOr,
+        new RelationalExpression(
+            Expression::Kind::kRelEQ,
+            new LabelAttributeExpression(new LabelExpression(*expr->arg()->from()),
+                                         new LabelExpression(*expr->arg()->prop())),
+            new ConstantExpression(Value(values[i]))), filter.get());
+        filter.reset(left);
+    }
+    return filter;
+}
+
+StatusOr<std::vector<std::string>> IndexScanValidator::textSearch(TextSearchExpression* expr) {
+    auto index = nebula::plugin::IndexTraits::indexName(space_.spaceDesc.space_name, isEdge_);
+    nebula::plugin::DocItem doc(index, *expr->arg()->prop(), schemaId_, *expr->arg()->val());
+    nebula::plugin::LimitItem limit(expr->arg()->timeout(), expr->arg()->limit());
+    std::vector<std::string> result;
+    // TODO (sky) : External index load balancing
+    for (auto& c : tsClients_) {
+        nebula::plugin::HttpClient ftc;
+        ftc.host = c.host;
+        if (c.__isset.user) {
+            ftc.user = c.user;
+        }
+        if (c.__isset.pwd) {
+            ftc.password = c.pwd;
+        }
+        bool suc = false;
+        switch (expr->kind()) {
+            case Expression::Kind::kTSFuzzy: {
+                folly::dynamic fuzz = folly::dynamic::object();
+                if (expr->arg()->fuzziness() == -1) {
+                    fuzz = "AUTO";
+                } else {
+                    fuzz = expr->arg()->fuzziness();
+                }
+                suc = nebula::plugin::ESGraphAdapter::kAdapter->fuzzy(ftc,
+                                                                      doc,
+                                                                      limit,
+                                                                      fuzz,
+                                                                      *expr->arg()->op(),
```

Comment:
What is op, and/or?

Reply:
> What is op, and/or?

Yes, same with match api of ElasticSearch.